sup3r/utilities/era_downloader.py

```diff
@@ -0,0 +1,1016 @@
+        """Convert z to given height variable
+
+        Parameters
+        ----------
+        standard_name : str
+            New variable name. e.g. 'zg' or 'orog'
+        long_name : str
+            Long name for new variable. e.g. 'Geopotential Height' or
+            'Orography'
+        old_ds : Dataset
+            Dataset() object from tmp file
+        ds : Dataset
+            Dataset() object for new file
+
+        Returns
+        -------
+        ds : Dataset
+            Dataset() object for new file with new height variable written.
+        """
+
+        _ = ds.createVariable(
+            standard_name, np.float32,
+            dimensions=old_ds['z'].dimensions)
+        ds.variables[standard_name][:] = old_ds['z'][:] / 9.81
+        ds.variables[standard_name].long_name = long_name
+        ds.variables[standard_name].standard_name = 'zg'
+        ds.variables[standard_name].units = 'm'
+        return ds
+
+    def process_level_file(self):
+        """Convert geopotential to geopotential height."""
+
+        dims = ('time', 'level', 'latitude', 'longitude')
+        tmp_file = self.get_tmp_file(self.level_file)
+        with Dataset(self.level_file, "r") as old_ds:
+            with Dataset(tmp_file, "w") as ds:
+                ds = self.init_dims(old_ds, ds, dims)
+
+                ds = self.convert_z('zg', 'Geopotential Height', old_ds, ds)
+
+                ds = self.map_vars(old_ds, ds)
+
+                if 'pressure' in self.variables:
+                    tmp = np.zeros(ds.variables['zg'].shape)
+                    for i in range(tmp.shape[1]):
+                        tmp[:, i, :, :] = ds.variables['level'][i] * 100
+                    _ = ds.createVariable('pressure', np.float32,
+                                          dimensions=dims)
+                    ds.variables['pressure'][:] = tmp[...]
+                    ds.variables['pressure'].long_name = 'Pressure'
+                    ds.variables['pressure'].units = 'Pa'
+
+        os.system(f'mv {tmp_file} {self.level_file}')
+        logger.info(f'Finished processing {self.level_file}. Moved '
+                    f'{tmp_file} to {self.level_file}.')
+
+    def process_and_combine(self):
+        """Process variables and combine."""
+
+        if not os.path.exists(self.combined_file) or self.overwrite:
+            logger.info(f'Processing {self.level_file}.')
+            self.process_level_file()
+            logger.info(f'Processing {self.surface_file}.')
+            self.process_surface_file()
+            logger.info(f'Combining {self.level_file} and {self.surface_file} '
+                        f'to {self.combined_file}.')
+            with xr.open_mfdataset([self.level_file, self.surface_file]) as ds:
+                ds.to_netcdf(self.combined_file)
+            logger.info(f'Finished writing {self.combined_file}')
+            os.remove(self.level_file)
+            os.remove(self.surface_file)
+
+
+class EraDownloader(BaseEraDownloader):
```

Comment:
why base/concrete classes? is BaseEraDownloader sub-classed somewhere else? Are there abstract methods?